pkg/registry/apis/dashboard/legacy/sql_dashboards.go

```diff
@@ -69,73 +75,100 @@ func NewDashboardAccess(sql db.DB, namespacer request.NamespaceMapper, dashStore
-	if !query.Labels.Empty() {
-		return nil, 0, fmt.Errorf("label selection not yet supported")
-	}
-	if len(query.Requirements.SortBy) > 0 {
-		return nil, 0, fmt.Errorf("sorting not yet supported")
-	}
-	if query.Requirements.ListHistory != "" {
-		return nil, 0, fmt.Errorf("ListHistory not yet supported")
-	}
-	if query.Requirements.ListDeleted {
-		return nil, 0, fmt.Errorf("ListDeleted not yet supported")
+const history = `SELECT
+	dashboard.org_id, dashboard.id,
+	dashboard.uid, dashboard.folder_uid,
+	dashboard.created,CreatedUSER.uid as created_by,
+	dashboard_version.created,CreatedUSER.uid as updated_by,
+	NULL, plugin_id,
+	dashboard_provisioning.name as origin_name,
+	dashboard_provisioning.external_id as origin_path,
+	dashboard_provisioning.check_sum as origin_key,
+	dashboard_provisioning.updated as origin_ts,
+	dashboard_version.version, dashboard_version.message, dashboard_version.data
+  FROM dashboard
+  LEFT OUTER JOIN dashboard_provisioning ON dashboard.id = dashboard_provisioning.dashboard_id
+  LEFT OUTER JOIN dashboard_version  ON dashboard.id = dashboard_version.dashboard_id
+  LEFT OUTER JOIN user AS CreatedUSER ON dashboard.created_by = CreatedUSER.id
+  LEFT OUTER JOIN user AS UpdatedUSER ON dashboard_version.created_by = UpdatedUSER.id
+  WHERE dashboard.is_folder = false`
+
+func (a *dashboardSqlAccess) getRows(ctx context.Context, query *DashboardQuery) (*rowsWrapper, int, error) {
+	if len(query.Labels) > 0 {
+		return nil, 0, fmt.Errorf("labels not yet supported")
+		// if query.Requirements.Folder != nil {
+		// 	args = append(args, *query.Requirements.Folder)
+		// 	sqlcmd = fmt.Sprintf("%s AND dashboard.folder_uid=$%d", sqlcmd, len(args))
+		// }
 	}
 
-	token, err := readContinueToken(query)
-	if err != nil {
-		return nil, 0, err
-	}
+	var sqlcmd string
+	args := []any{query.OrgID}
 
 	limit := query.Limit
 	if limit < 1 {
 		limit = 15 //
 	}
-	args := []any{query.OrgID}
 
-	sqlcmd := selector
+	if query.GetHistory {
+		if query.GetTrash {
+			return nil, 0, fmt.Errorf("trash not included in history table")
+		}
 
-	// We can not do this yet because title + tags are in the body
-	if onlySummary && false {
-		sqlcmd = strings.Replace(sqlcmd, "dashboard.data", `"{}"`, 1)
-	}
+		sqlcmd = fmt.Sprintf("%s AND dashboard.org_id=$%d\n  ", history, len(args))
+
+		if query.UID == "" {
+			return nil, 0, fmt.Errorf("history query must have a UID")
+		}
 
-	sqlcmd = fmt.Sprintf("%s AND dashboard.org_id=$%d", sqlcmd, len(args))
-	if query.UID != "" {
 		args = append(args, query.UID)
 		sqlcmd = fmt.Sprintf("%s AND dashboard.uid=$%d", sqlcmd, len(args))
+
+		if query.Version > 0 {
+			args = append(args, query.Version)
+			sqlcmd = fmt.Sprintf("%s AND dashboard_version.version=$%d", sqlcmd, len(args))
+		} else if query.MinID > 0 {
+			args = append(args, query.MinID)
+			sqlcmd = fmt.Sprintf("%s AND dashboard_version.version<$%d", sqlcmd, len(args))
```

Comment:
Shouldn't it be >= if it's the minimum id?

Reply:
hymm -- maybe a better variable name.  The MinID is the last ID that was sent, so we want the next one

Reply:
I updated the variable name to `LastID` -- it is the last ID sent.. so should be skipped